src/main.js

```diff
@@ -49,6 +49,23 @@ var mainWindow = null;
 var trayIcon = null;
 var willAppQuit = false;
 
+if (config.httpBasicAuth) {
+
+  var readlineSync = require('readline-sync');
+
+  var username = readlineSync.question('Username: ');
+  var password = readlineSync.question('Password: ', {
+    hideEchoBack: true // The typed text on screen is hidden by `*` (default). 
+  });
+
+  app.on('login', function(event, webContents, request, authInfo, callback) {
+    event.preventDefault();
+    console.log("Replacing default auth behaviour.");
```

Comment:
Why don't you use the prompt in the 'login' event? It appears when it's needed.

In this commit, the app doesn't show anything when the app is executed by double-click. Users must know that command-line is necessary to use this feature.